resources/nspawn.go

```diff
@@ -115,8 +177,9 @@ func (obj *NspawnRes) Watch() error {
 	if err := call.Err; err != nil {
 		return err
 	}
-	buschan := make(chan *dbus.Signal, 10)
-	bus.Signal(buschan)
+	// TODO: verify that dbus doesn't drop messages without a chan buffer
```

Comment:
I think I was wrong. I'm sorry. I think this needs to be buffered to reduce the chance of deadlock due to issues in the godbus implementation. So please add the buffer back and a relevant comment asking about if the implementation can deadlock if we close when there's still a message not read from the channel.

Alternatively, the harder thing to do is to make sure we drain the channel on close-- I think it's probably better we leave that to a future patch. I want to merge this thing.